File: src/main/java/io/playground/orderservice/domain/order/Order.java
package io.playground.orderservice.domain.order;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class Order {
    private Long id;

    private String externalId;

    private Long userId;

    private OrderStatus status;

    public enum OrderStatus {
        CREATED, PAID, COMPLETED,
        CANCELED, PARTIAL_CANCELED, FAILED
    }

    public static Order of(Long id,
                           String externalId,
                           Long userId,
                           OrderStatus status) {
        return new Order(id, externalId, userId, status);
    }
}

File: src/main/kotlin/io/playground/orderservice/domain/order/Order.kt
package io.playground.orderservice.domain.order

class Order(
    val id: Long?,
    val externalId: String,
    val userId: Long?,
    var status: OrderStatus,
) {
    enum class OrderStatus {
        CREATED,
        PAID,
        COMPLETED,
        CANCELED,
        PARTIAL_CANCELED,
        FAILED,
    }

    companion object {
        @JvmStatic
        fun of(
            id: Long?,
            externalId: String,
            userId: Long?,
            status: OrderStatus,
        ): Order = Order(id, externalId, userId, status)
    }
}
